src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferStreamReadAsyncMemoryOverloadsTests.cs

```diff
@@ -0,0 +1,558 @@
+        End Using
+    End Sub
+End Class
+            ");
+        }
+
+        [Fact]
+        public async Task VB_Analyzer_NoDiagnostic_ContinueWith_ContinueWith_ConfigureAwait()
+        {
+            await AnalyzeVBAsync(@"
+Imports System
+Imports System.IO
+Imports System.Threading
+Class C
+    Public Async Sub M()
+        Using fs As FileStream = File.Open(""file.txt"", FileMode.Open)
+            Dim buffer As Byte() = New Byte(fs.Length - 1) {}
+            Await fs.ReadAsync(buffer, 0, fs.Length).ContinueWith(Sub(c)
+                                                                        End Sub).ContinueWith(Sub(c)
+                                                                                              End Sub).ConfigureAwait(False)
+        End Using
+    End Sub
+End Class
+            ");
+        }
+
+        [Fact]
+        public async Task VB_Analyzer_NoDiagnostic_CancellationToken_ConfigureAwait()
+        {
+            await AnalyzeVBAsync(@"
+Imports System
+Imports System.IO
+Imports System.Threading
+Class C
+    Public Async Sub M()
+        Using fs As FileStream = File.Open("", file.txt, "", FileMode.Open)
+            Dim buffer As Byte() = New Byte(fs.Length - 1) {}
+            Await fs.ReadAsync(buffer, 0, CInt(fs.Length), New CancellationToken()).ConfigureAwait(False)
+        End Using
+    End Sub
+End Class
+            ");
+        }
+
+        [Fact]
+        public async Task VB_Analyzer_NoDiagnostic_UnsupportedVersion()
+        {
+            await AnalyzeVBUnsupportedAsync(@"
+Imports System
+Imports System.IO
+Imports System.Threading
+Class C
+    Public Async Sub M()
+        Using fs As FileStream = File.Open(""file.txt"", FileMode.Open)
+            Dim buffer As Byte() = New Byte(fs.Length - 1) { }
+            Await fs.ReadAsync(buffer, 0, fs.Length)
+        End Using
+    End Sub
+End Class
+            ");
+        }
+
+        #endregion
+
+        #region C# - Diagnostic - Analyzer
+
+        [Fact]
+        public async Task CS_Analyzer_Diagnostic_ByteArray()
+        {
+            await AnalyzeCSAsync(@"
+using System;
+using System.IO;
+using System.Threading;
+class C
+{
+    public async void M()
+    {
+        using (FileStream fs = File.Open(""file.txt"", FileMode.Open))
+        {
+            byte[] buffer = new byte[fs.Length];
+            await fs.ReadAsync(buffer, 0, (int)fs.Length);
+        }
+    }
+}
+            ", GetCSResult(12, 19));
+        }
+
+        [Fact]
+        public async Task CS_Analyzer_Diagnostic_Inline()
+        {
+            await AnalyzeCSAsync(@"
+using System;
+using System.IO;
+using System.Threading;
+class C
+{
+    public async void M()
+    {
+        using (FileStream fs = File.Open(""file.txt"", FileMode.Open))
+        {
+            await fs.ReadAsync(new byte[fs.Length], 0, (int)fs.Length);
+        }
+    }
+}
+            ", GetCSResult(11, 19));
```

Comment:
I believe that the current requested style is that if you can't use the builtin markup that you use the Span instead of Position (e.g. the 4-ple instead of the 2-ple).

Reply:
Changed to span.